### PR TITLE
IGNITE-16802 Shared RocksDB instance for the Raft Log Storage

### PR DESCRIPTION
**Pull request by SammyVimes**
https://issues.apache.org/jira/browse/IGNITE-16802

### CODE REVIEW

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/util/Utils.java [hunk 1]
```diff
@@ -123,6 +126,29 @@ public static Future<?> runClosureInThread(ExecutorService executor, final Closu
         return runClosureInThread(executor, done, Status.OK());
     }
 
+    private static final long pid = ProcessHandle.current().pid();
```
Comment (sk0x50) [inline modules/raft/src/main/java/org/apache/ignite/raft/jraft/util/Utils.java:129]: This looks like a debug piece of code. Should it be removed?

File: modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java [hunk 2]
```diff
@@ -157,10 +157,10 @@ public void stop() {
                     Thread.currentThread().interrupt();
 
                     throw new IgniteInternalException("Interrupted while waiting for the ClusterService to stop", e);
-                } catch (TimeoutException e) {
-                    throw new IgniteInternalException("Timeout while waiting for the ClusterService to stop", e);
                 } catch (ExecutionException e) {
                     throw new IgniteInternalException("Unable to stop the ClusterService", e.getCause());
+                } catch (TimeoutException ignored) {
+                    // Failed to leave gracefully
```
Comment (sk0x50) [inline modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java:163]: Perhaps, it makes sense to log a warning message here.

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/JRaftServiceFactory.java [hunk 3]
```diff
@@ -31,11 +31,11 @@ public interface JRaftServiceFactory {
     /**
      * Creates a raft log storage.
      *
-     * @param uri The log storage uri from {@link NodeOptions#getSnapshotUri()}
+     * @param groupId The name of the raft group.
      * @param raftOptions the raft options.
      * @return storage to store raft log entires.
      */
-    LogStorage createLogStorage(final String uri, final RaftOptions raftOptions);
+    LogStorage createLogStorage(final String groupId, final RaftOptions raftOptions);
```
Comment (sk0x50) [inline modules/raft/src/main/java/org/apache/ignite/raft/jraft/JRaftServiceFactory.java:38]: Everything else works with the URI. Why is this change needed? It doesn't look very consistent.

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/util/Utils.java (hunk 1 above)
Comment (SammyVimes) [reply modules/raft/src/main/java/org/apache/ignite/raft/jraft/util/Utils.java:129]: You're right, I completely forgot about this

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/JRaftServiceFactory.java (hunk 3 above)
Comment (SammyVimes) [reply modules/raft/src/main/java/org/apache/ignite/raft/jraft/JRaftServiceFactory.java:38]: All log storages have the same URI in this PR. The only way to distinguish data in a storage is by raft group id. Hence the change

File: modules/raft/src/test/java/org/apache/ignite/raft/jraft/storage/impl/SharedVsNonSharedLogStorageBenchmark.java [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+import org.apache.ignite.raft.jraft.option.RaftOptions;
+import org.apache.ignite.raft.jraft.storage.LogStorage;
+import org.apache.ignite.raft.jraft.storage.LogStorageProvider;
+import org.apache.ignite.raft.jraft.test.TestUtils;
+import org.apache.ignite.raft.jraft.util.Utils;
+
+/**
+ * Benchmark for shared versus non-shared log storage.
+ */
+public class SharedVsNonSharedLogStorageBenchmark {
```
Comment (sashapolo) [inline modules/raft/src/test/java/org/apache/ignite/raft/jraft/storage/impl/SharedVsNonSharedLogStorageBenchmark.java:46]: please fix all code style issues in this class

**Review by sashapolo — APPROVED** [review]
